File: patch_tests.js
const fs = require('fs');
const file = 'asguard-interceptor/tests/interceptor.test.ts';
let content = fs.readFileSync(file, 'utf8');

// The tests for "POST /admin/anomaly/triage" need to assert the audit log writes correctly.

// Update block test
const blockTestSearch = `
      const blacklistRecord = await env.ASGUARD_BLACKLIST.get("ip:1.2.3.4");
      expect(blacklistRecord).toBe("1");

      const anomalyQueue = await env.ASGUARD_TELEMETRY.get("anomaly_queue");
      expect(anomalyQueue).toBeNull();
    });`;

const blockTestReplace = `
      const blacklistRecord = await env.ASGUARD_BLACKLIST.get("ip:1.2.3.4");
      expect(blacklistRecord).toBe("1");

      const anomalyQueue = await env.ASGUARD_TELEMETRY.get("anomaly_queue");
      expect(anomalyQueue).toBeNull();

      const auditCalls = customMockTelemetry.put.mock.calls.filter((c: any) => c[0].startsWith("audit:") && c[1].includes("onyx_anomaly_triaged"));
      expect(auditCalls.length).toBe(1);
      const auditPayload = JSON.parse(auditCalls[0][1]);
      expect(auditPayload.decision).toBe("block");
      expect(auditPayload.target).toBe("1.2.3.4");
      expect(auditPayload.authorizedByWallet).toBe("test_api_key_123");
    });`;

if (content.includes(blockTestSearch)) {
  content = content.replace(blockTestSearch, blockTestReplace);
} else {
  console.log("Could not find block test search string.");
}

// Update dismiss test
const dismissTestSearch = `
      const blacklistRecord = await env.ASGUARD_BLACKLIST.get("ip:5.5.5.5");
      expect(blacklistRecord).toBeNull(); // Ensure it didn't block
    });`;

const dismissTestReplace = `
      const blacklistRecord = await env.ASGUARD_BLACKLIST.get("ip:5.5.5.5");
      expect(blacklistRecord).toBeNull(); // Ensure it didn't block

      const auditCalls = customMockTelemetry.put.mock.calls.filter((c: any) => c[0].startsWith("audit:") && c[1].includes("onyx_anomaly_triaged"));
      expect(auditCalls.length).toBe(1);
      const auditPayload = JSON.parse(auditCalls[0][1]);
      expect(auditPayload.decision).toBe("dismiss");
      expect(auditPayload.target).toBe("5.5.5.5");
      expect(auditPayload.authorizedByWallet).toBe("sig123");
    });`;

if (content.includes(dismissTestSearch)) {
  content = content.replace(dismissTestSearch, dismissTestReplace);
} else {
  console.log("Could not find dismiss test search string.");
}

fs.writeFileSync(file, content, 'utf8');
console.log('Done patching tests');

File: patch_tests.py
import re

with open('asguard-interceptor/tests/interceptor.test.ts', 'r') as f:
    content = f.read()

new_test = """
  it("rejects autonomous block if key targets protected internal namespace", async () => {
    const protectedKeys = ["token:axim_core_secret", "ip:10.0.0.1", "ip:127.0.0.1", "ip:192.168.1.1"];

    for (const key of protectedKeys) {
      const request = new Request("http://localhost/blocklist/autonomous", {
        method: "POST",
        headers: {
          "Authorization": "Bearer test-ai-mutation-key",
        },
        body: JSON.stringify({
          key,
          ttl: 3600
        }),
      });
      const ctx = { waitUntil: vi.fn() } as any;
      const response = await worker.fetch(request, env as any, ctx);
      expect(response.status).toBe(400);
      expect(await response.text()).toContain("protected internal target");
    }
  });
"""

content = content.replace('it("rejects autonomous block with invalid token", async () => {', new_test + '\n  it("rejects autonomous block with invalid token", async () => {')

with open('asguard-interceptor/tests/interceptor.test.ts', 'w') as f:
    f.write(content)
